### ISSUE TITLE
Unable to load right_ptr from the cursor theme

### ISSUE THREAD

**Issue by tekstryder** [geany/geany#3809]
- Arch Linux | Kernel 6.7.11
- Gnome-shell | Mutter 46.0
- Wayland (*meson_options: xwayland=false*)
- `Geany 2.0`
- Gtk 3.24.41

Seeing these errors since Gnome 46.

```
Mar 30 08:00:22 geany[10386]: Unable to load right_ptr from the cursor theme
Mar 30 08:00:22 geany[10386]: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
Mar 30 12:52:13 geany[15506]: Unable to load right_ptr from the cursor theme
Mar 30 12:52:13 geany[15506]: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
```

For context, see the relevant issues/merges/discussions for the whole Adwaita cursor kerfuffle this past dev cycle.

- https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/issues/278
- https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/3295
- https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/commit/74e9b79471236320d2af4925d6c5bb7df22380ce
- https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/merge_requests/65

**Comment by elextr** [comment]
Well don't use dead tortoise as your theme is likely the only answer Geany can give, we don't don't ship or set icon themes.

**Comment by tekstryder** [comment]
> we don't don't ship or set icon themes

No, but you utilize cursors.

I suppose this is more a request for Geany supporting CSS xcursor standards[1][2] than finding icon themes that support esoteric cursors. 

I'm not sure what the intended use of `right_ptr` is by Geany. What is that supposed to indicate to the user?

[1] https://www.w3.org/TR/css-ui-4/#cursor
[2] https://developer.mozilla.org/en-US/docs/Web/CSS/cursor

**Comment by b4n** [comment]
I'll have to check the details from the OP, but all custom cursors are probably coming from Scintilla. IIRC it uses a right pointed cursor over the left margin.

Le 31 mars 2024 13:54:22 GMT+02:00, tekstryder ***@***.***> a écrit :

**Comment by tekstryder** [comment]
> IIRC it uses a right pointed cursor over the left margin

Ah, you're right (pun intended). Still unclear what that intends to indicate, but that's where the `right_ptr` attempt occurs.

**Comment by elextr** [comment]
Yes, Scintilla can use the cursors [here](https://github.com/geany/geany/blob/3a92129d61d4d8e96136e29d4709bb638527be40/scintilla/gtk/PlatGTK.cxx#L1297) and it just asks GDK for them.

It is [documented](https://www.scintilla.org/ScintillaDoc.html#SCI_SETMARGINCURSORN) that a right cursor is shown over margins and Geany does not set it, just leaves it as default.  A PR making a setting to select a different cursor would probably be accepted so long as default was as is.

Elsewhere Scintilla notes use of another cursor icon for clickable links, not used by Geany, but I think plugin(s) might use it.

A quick look at the discussion the OP linked seems to suggest that the problem is being fixed by linking under the old names as well as the new names, so I doubt its worth it doing anything in Scintilla or Geany.  The problem should only affect bleeding edge rolling distros for a short time, and if it doesn't get in a release should not appear in stable distros.

**Comment by elextr** [comment]
> > if it doesn't get in a release should not appear in stable distros.

> No, that "compromise" [merged](https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/merge_requests/65) during the past dev cycle, and was included in the tagged release. All rolling/current distros that default to Adwaita are missing right_ptr.

Sorry, if the linking of the old names compromise was in the release why is right_ptr missing, or did they forget that one?

**Comment by tekstryder** [comment]
> why is right_ptr missing, or did they forget that one?

Possibly, but there are other [more hotly debated](https://gitlab.gnome.org/GNOME/gtk/-/issues/6337) one's missing/undecided too, such as left_ptr_watch and various dnd.

I only know enough to see that right-ptr was axed in the [initial change](https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/merge_requests/61/diffs?commit_id=74e9b79471236320d2af4925d6c5bb7df22380ce), which was later partially reverted by what I referred to as a 'compromise'... albeit sans the right_ptr.

I'm not a gnome/gtk affiliate, just a curious user following the fallout from this change, so I've filed issues with a few affected applications such as here, [Evolution](https://gitlab.gnome.org/GNOME/evolution/-/issues/2708#note_2067549) Mail, and [Foot](https://codeberg.org/dnkl/foot/issues/1624#issuecomment-1624509) terminal. The Foot developer wasn't too keen on this change either but did make some [quick accommodations for preferring CSS cursor names](https://codeberg.org/dnkl/foot/pulls/1626).

I really never knew how many folks were so passionate about cursors! I don't care so much about my pointer, but I do cherish a clean system journal.

**Comment by elextr** [comment]
> I really never knew how many folks were so passionate about cursors!

I'm not sure its so much about cursors as it is G* folks going back on their word and breaking GTK3 applications _again_, just proves that we can't trust 'em, Leopards never change their spots etc etc.

Oh well, my answer is "Don't use dead tortoises as themes, make a proper one you lazy ...." :grin:

Basically a "default" theme is never going to be right for anyone, so it should be as lousy as possible (like default HTML) which Adwaita achieves IMNSHO so all distros should make their own and not be lazy [end rant]

I don't know why Scintilla uses different cursors, possibly to match the behaviour of the initial Windows control it is based on.  It tries to be consistent on all platforms.

Perhaps Scintilla should switch to checking the return [here](https://github.com/geany/geany/blob/3a92129d61d4d8e96136e29d4709bb638527be40/scintilla/gtk/PlatGTK.cxx#L1297) and  and falling back to default if fail.

**Comment by satmandu** [comment]
Hello all, I'm also seeing this in Ubuntu 25.04:
```
 Gdk-Message: 12:17:02.285: Unable to load right_ptr from the cursor theme
```

I (and I assume most ubuntu users) are using the default Adwaita themes that come with Ubuntu.

Is there anything we can do to resolve this error message?

**Comment by tekstryder** [comment]
Wow, as reporter, I've long since been on Sway and Hyprland where this was no longer an issue.

But, very cool a fix has been merged. Thanks!
